source/slang/parser.cpp

```diff
@@ -1452,9 +1462,16 @@ namespace Slang
 
         RefPtr<Expr> typeExpr = basicType;
 
-        if (parser->LookAheadToken(TokenType::OpLess))
+        while (parser->LookAheadToken(TokenType::OpLess) || parser->LookAheadToken(TokenType::Dot))
```

Comment:
Minor style issue, but in cases like this where things change from a single lookahead case we care about to multiple, I usually write something like:

```
for(;;) // forever
{
    switch(peekTokenCode(parser))
    {
    case TokenType::OpLess:
        { ... }
        continue;

    case TokenType:Dot:
        { ... }
        continue;

    default:
        break;
    }
    break;
}
```

The basic rationale being both to avoid long `if`/`else` chains and favor `switch` dispatch, and to avoid the duplication of the token types we care about in two different locations in the code.

It's mostly just a style thing, but I find it makes the code easier to maintain as cases get added/removed/changed (that is part of the reason why you'll see me use `switch` statements even when I only care about two cases).